docs/hardening.md

```diff
@@ -0,0 +1,112 @@
+
+To apply the hardening configuration, create a file (eg. `hardening.yaml`) and paste the content of the following code snippet into that.
+
+**N.B.** Some of these configurations have just been added to **Kubespray**, so ensure that you have the latest version to make it works properly.
+
+Also, ensure that other configurations doesn't override these.
+
+`hardening.yaml`:
+
+```yaml
+# Hardening
+---
+# This allow us to use PodSecurity admission plugin which is provided as built-in
+kube_version: v1.23.6
+
+## kube-apiserver
+authorization_modes: ['Node','RBAC']
+kube_apiserver_feature_gates: ['AppArmor=true']
+kube_apiserver_request_timeout: 120s
+kube_apiserver_service_account_lookup: true
+
+# enable kubernetes audit
+kubernetes_audit: true
+audit_log_path: "/var/log/kube-apiserver-log.json"
+audit_log_maxage: 30
+audit_log_maxbackups: 10
+audit_log_maxsize: 100
+
+tls_min_version: VersionTLS12
+tls_cipher_suites:
+  - TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
+  - TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
+  - TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
+
+# enable encryption at rest
+kube_encrypt_secret_data: true
+kube_encryption_resources: [secrets]
+kube_encryption_algorithm: "secretbox"
+
+kube_apiserver_enable_admission_plugins: ['EventRateLimit,AlwaysPullImages,ServiceAccount,NamespaceLifecycle,NodeRestriction,LimitRanger,ResourceQuota,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,PodNodeSelector,PodSecurity']
+kube_apiserver_admission_control_config_file: true
+# EventRateLimit plugin configuration
+kube_apiserver_admission_event_rate_limits:
+  limit_1:
+    type: Namespace
+    qps: 50
+    burst: 100
+    cache_size: 2000
+  limit_2:
+    type: User
+    qps: 50
+    burst: 100
+
+kube_profiling: false
+kube_proxy_profiling: false
+
+## kube-controller-manager
+kube_controller_manager_bind_address: 127.0.0.1
+kube_controller_terminated_pod_gc_threshold: 50
+kube_controller_feature_gates: ["RotateKubeletServerCertificate=true","AppArmor=true"]
+
+## kube-scheduler
+kube_scheduler_bind_address: 127.0.0.1
+kube_kubeadm_scheduler_extra_args:
+  profiling: false
+kube_scheduler_feature_gates: ["AppArmor=true"]
+
+## etcd
+etcd_deployment_type: kubeadm
+
+## kubelet
+kubelet_authorization_mode_webhook: true
+kubelet_authentication_token_webhook: true
+kube_read_only_port: 0
+kubelet_rotate_server_certificates: true
+kubelet_protect_kernel_defaults: "true"
+kubelet_event_record_qps: 1
+kubelet_rotate_certificates: "true"
+kubelet_streaming_connection_idle_timeout: "5m"
+kubelet_make_iptables_util_chains: true
+kubelet_feature_gates: ["RotateKubeletServerCertificate=true"]
+```
+
+Let's take a deep look to the resultant **kubernetes** configuration:
+
+* First of all, we have choosen to set the **kubernetes** version to `v1.23.6` to have all the most recent security features (eg. the new `PodSecurity` admission plugin).
```

Comment:
ditt: Please consider readers could change here to the specific `v1.23.6` even if the default kube_version is a newer one, if we miss updating this document.

Reply:
Agree. Maybe I can write that the `1.23.6` is the minimum version to use this hardening configuration. WDYT?

Reply:
> Agree. Maybe I can write that the `1.23.6` is the minimum version to use this hardening configuration. WDYT?

That looks good for me, thank you for your suggestion :smile: